From ebc8bf4a58f7f65dbb7e7b2e1152322159faa228 Mon Sep 17 00:00:00 2001
From: Cathy Avery <cavery@redhat.com>
Date: Fri, 1 Jun 2018 15:58:51 +0200
Subject: [PATCH] Ignore ENXIO errors with SyncDriver

RH-Author: Cathy Avery <cavery@redhat.com>
Message-id: <20180601155851.7526-1-cavery@redhat.com>
Patchwork-id: 80546
O-Subject: [RHEL7.5.z open-vm-tools PATCH] Ignore ENXIO errors with SyncDriver
Bugzilla: 1582123
RH-Acked-by: Richard Jones <rjones@redhat.com>
RH-Acked-by: Mohammed Gamal <mgamal@redhat.com>
RH-Acked-by: Eduardo Otubo <otubo@redhat.com>

A quiesced snapshot fails when the target VM has a bind mount.  The
problem is that the syncDriver gets an ENXIO error when it tries to
open the mount point.  To fix the problem, let the quiesce operation
skip a mount point when opening it results in an ENXIO error.

This change is based on pull request
https://github.com/vmware/open-vm-tools/pull/218 .

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

Conflicts:
           open-vm-tools/AUTHORS version 10.1.10 is empty.

(cherry picked from commit 6559b7dfd9f6f40c2df76e1b1e785fa3eb1c1da4)
Signed-off-by: Cathy Avery <cavery@redhat.com>
---
 lib/syncDriver/syncDriverLinux.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/open-vm-tools/lib/syncDriver/syncDriverLinux.c b/open-vm-tools/lib/syncDriver/syncDriverLinux.c
index da0a8cb..4422729 100644
--- a/lib/syncDriver/syncDriverLinux.c
+++ b/lib/syncDriver/syncDriverLinux.c
@@ -1,5 +1,5 @@
 /*********************************************************
- * Copyright (C) 2011-2017 VMware, Inc. All rights reserved.
+ * Copyright (C) 2011-2018 VMware, Inc. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -195,6 +195,14 @@ LinuxDriver_Freeze(const GSList *paths,
             Debug(LGPFX "cannot access mounted directory '%s'.\n", path);
             continue;
 
+         case ENXIO:
+            /*
+             * A bind-mounted file, such as a mount of /dev/log for a
+             * chrooted application, will land us here.  Just skip it.
+             */
+            Debug(LGPFX "no such device or address '%s'.\n", path);
+            continue;
+
          case EIO:
             /*
              * A mounted HGFS filesystem with the backend disabled will give
-- 
1.8.3.1

